crates/fuel-core/src/service/sub_services/algorithm_updater.rs

```diff
@@ -339,23 +335,8 @@ where
 
         updater.update_l2_block_data(height, block_gas_used, block_gas_capacity)?;
         let metadata = AlgorithmUpdater::V0(updater.clone()).into();
-        metadata_storage.set_metadata(metadata)?;
+        metadata_storage.set_metadata(&metadata)?;
     }
 
     Ok(())
 }
-
-fn revert_gas_price_db_to_height(
```

Comment:
why is this fragment of code removed? Is it related to the temporary fix in `combined_database.rs`?

Reply:
yes we don't need this code anymore since we delegate rolling back of db to `FuelService`